src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestsData.cs

```diff
@@ -0,0 +1,2160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.XUnitExtensions;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    public class MLDsaNistTestCase
+    {
+        public int NistTestCaseId { get; init; }
+        public MLDsaAlgorithm Algorithm { get; init; }
+        public bool ShouldPass { get; init; }
+
+        public string PublicKeyHex { get; init; }
+        public string SecretKeyHex { get; init; }
+
+        public string MessageHex { get; init; }
+        public string ContextHex { get; init; }
+        public string SignatureHex { get; init; }
+
+        public byte[] PublicKey => PublicKeyHex.HexToByteArray();
+        public byte[] SecretKey => SecretKeyHex.HexToByteArray();
+
+        public byte[] Message => MessageHex.HexToByteArray();
+        public byte[] Context => ContextHex.HexToByteArray();
+        public byte[] Signature => SignatureHex.HexToByteArray();
+    }
```

Comment:
IIRC, Xunit will invent a ToString for this object equal to all of the properties and their values.

It's probably goodness to do something like

```C#
public override string ToString() => $"NIST Test Case {NistTestCaseId} (Alg: {Algorithm})";
```

Make one of the tests using this as a Theory parameter fail, then add the ToString, and tweak it until it looks nice.